assets/src/edit-story/utils/useIntersectionEffect.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffect } from 'react';
+const IntersectionObserver = require('intersection-observer-polyfill/dist/IntersectionObserver');
+
+/**
+ * @param {!{current: ?Element}} ref Target node ref.
+ * @param {function(IntersectionObserverEntry)} handler The intersection
+ * handler.
+ * @param {?Object} options The IntersectionObserver options.
+ * @param {!Array=} deps The effect's dependencies.
+ */
+function useIntersectionEffect(ref, handler, options = {}, deps = undefined) {
+  useEffect(
+    () => {
+      const node = ref.current;
+      const usingRoot = 'root' in options;
+      if (usingRoot) {
+        options.root = options.root.current;
```

Comment:
Any reason for not passing the `options.root.current` directly into the options?

Reply:
[here?](https://github.com/google/web-stories-wp/pull/562/files/ae012fe31aa7984db6acc3ba812dacf97b78f331#diff-468f2d4f5a0260eca778fdf38e9a9a3bR238) if yes - I just followed the same convention as in `useResizeEffect`, it takes `ref`, should we only operate on DOM nodes and only take refs if we intend to modify them?

Reply:
Yes, there. Just that the naming is a bit ambiguous -- `root` seems to be a node by name, however, it's actually passing a ref. I was wondering that maybe you didn't pass the node directly for some specific reason, to skip the `options.root = options.root.current` thing here. There's probably no difference if to pass a ref or a node directly, however, it would be good to match the naming.

Reply:
Good call, also this was not quite true
```js
* @param {?Object} options The IntersectionObserver options.
```
Done in https://github.com/google/web-stories-wp/pull/562/commits/715f2a730221c493f47635ac42fc53001fd10b1f